src/app.tsx

```diff
@@ -17,34 +17,32 @@
  * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
  */
 
-import cockpit from 'cockpit';
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Alert } from "@patternfly/react-core/dist/esm/components/Alert/index.js";
 import { Card, CardBody, CardTitle } from "@patternfly/react-core/dist/esm/components/Card/index.js";
 
+import cockpit from 'cockpit';
+
 const _ = cockpit.gettext;
 
-export class Application extends React.Component {
-    constructor() {
-        super();
-        this.state = { hostname: _("Unknown") };
+export const Application = () => {
+    const [hostname, setHostname] = useState(_("Unknown"));
 
-        cockpit.file('/etc/hostname').watch(content => {
-            this.setState({ hostname: content.trim() });
-        });
-    }
+    useEffect(() => {
+        const hostname = cockpit.file('/etc/hostname');
+        hostname.watch(content => setHostname(content?.trim() ?? ""));
+        return hostname.close;
+    });
```

Comment:
This doesn't look right. I think it should have a empty dependency array to limit it to only run once after mount.

Reply:
@subhoghoshX Oh, you are right! Fixed in #920.

... and nice to see you again! :wave:

Reply:
Hey, haha I just opened a PR. Maybe I should close it.